spec/contracts/projects/update_contract_spec.rb

```diff
@@ -33,22 +33,24 @@
 
 RSpec.describe Projects::UpdateContract do
   it_behaves_like "project contract" do
-    let(:custom_field) do
-      build_stubbed(:integer_project_custom_field).tap do |cf|
-        cf.id = "1001"
-        allow_any_instance_of(CustomValue) # rubocop:disable RSpec/AnyInstance
-          .to receive(:custom_field).and_return(cf)
-      end
-    end
+    shared_let(:custom_field) { create(:integer_project_custom_field) }
+    shared_let(:admin_only_custom_field) { create(:integer_project_custom_field, admin_only: true) }
+    shared_let(:not_enabled_custom_field) { create(:integer_project_custom_field) }
 
     let(:project) do
       build_stubbed(:project,
                     active: project_active,
                     public: project_public,
                     status_code: project_status_code,
                     status_explanation: project_status_explanation).tap do |p|
-        allow(p).to receive_messages(available_custom_fields: [custom_field],
-                                     all_available_custom_fields: [custom_field])
+        # Use real AR relations for the custom field associations with actual IDs
+        available_custom_fields_relation = ProjectCustomField.where(id: [custom_field.id, admin_only_custom_field.id])
+        all_available_custom_fields_relation = ProjectCustomField.where(id: [custom_field.id, admin_only_custom_field.id,
+                                                                             not_enabled_custom_field.id])
```

Comment:
`.id` can be skipped:
```suggestion
        available_custom_fields_relation = ProjectCustomField.where(id: [custom_field, admin_only_custom_field])
        all_available_custom_fields_relation = ProjectCustomField.where(id: [custom_field, admin_only_custom_field,
                                                                             not_enabled_custom_field])
```